components/datetime/src/options/components.rs

```diff
@@ -0,0 +1,209 @@
+use crate::fields::{self, Field, FieldLength, FieldSymbol, FieldType};
+use std::fmt;
+
+#[derive(Debug)]
+pub struct Bag {
+    pub era: Text,
+    pub year: Numeric,
+    pub month: Month,
+    pub day: Numeric,
+    pub weekday: Text,
+
+    pub hour: Numeric,
+    pub minute: Numeric,
+    pub second: Numeric,
+    pub hour_cycle: HourCycle,
+
+    pub time_zone_name: TimeZoneName,
+}
+
+impl Default for Bag {
+    fn default() -> Self {
+        Self {
+            era: Text::default(),
+            year: Numeric::Numeric,
+            month: Month::Long,
+            day: Numeric::Numeric,
+            weekday: Text::default(),
+
+            hour: Numeric::Numeric,
+            minute: Numeric::Numeric,
+            second: Numeric::Numeric,
+            hour_cycle: HourCycle::default(),
+
+            time_zone_name: TimeZoneName::default(),
+        }
+    }
+}
+
+impl Bag {
+    pub fn write_skeleton(&self, w: &mut impl fmt::Write) -> fmt::Result {
```

Comment:
I'm scratching my head over this function.  What is its intended purpose?  Is it writing out a real UTS 35 datetime skeleton?  How come the implementation only looks at era, year, month, day, and hour, but not the rest of the fields?

Reply:
Yes, this is supposed to produce UTF25 datetime skeleton.
This is just the state of the WIP, my intention was to explore how this can be done, and I suggest the following design. The rest of the fields will be added to `get`.

Is the completness your only concern?